### PR TITLE
docs: add special_route.md

### PR DESCRIPTION
**Pull request by DammneD**

### CODE REVIEW

File: docs/routing/special_routes.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Special routes
+
+To see special routes, use the `Show special` switcher at the top of the routes page
+
+It is used to render errors, if an application with the `Primary` or `Essential` type falls on the page, ILC will not find a match between the address bar and `Route`, `Template` will not be specified in the route chain, etc., ILC will use `Special route`.
```
Comment (StyleT) [inline docs/routing/special_routes.md:5]: There are no "special route" for 5XX err technically

File: docs/routing/special_routes.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+- `Special route` must have a `Template`.
+
+- `Special route` must be created separately for each domain.
+
+- More about [404 Error](../global_errors_handling.md#404-error-not-found)
+
+### Example:
+
+Let's go to the `/wrapper/blablabla/` route, ILC will render [404 Error](../global_errors_handling.md#404-error-not-found), because the specified route does not exactly match our routes, аnd all other routes (like `*`) will be ignored (in this case `navbar` it's part of [404 Error](../global_errors_handling.md#404-error-not-found), **not** our route `*` with `Order pos - 100`).
```
Comment (StyleT) [inline docs/routing/special_routes.md:27]: It's unclear from text why "*" route will be ignored

File: docs/routing/special_routes.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+![ILC registry route to error](../assets/routes/route.png)
+
+**Result:**
+
+![ILC registry 404 error example](../assets/routes/fourth-case-result.png)
+
+## '500' Error
+
+This error occurs if the ILC is unable to process the request (for example, because no [template](./route_configuration_options.md#route_template) is specified in the route chain, or other problems related to the inability of ILC to render the page).
```
Comment (StyleT) [inline docs/routing/special_routes.md:37]: Pls give a link to err handling documentation so users can see more details

File: docs/routing/special_routes.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+![ILC registry 500 Error](../assets/routes/500-error.png)
+
+- This error is a simple HTML template.
+
+- It is designed in such a way that ILC can render it under any conditions.
+
+- More about [500 Error](../global_errors_handling.md#5xx-errors-unexpected-errors).
```
Comment (StyleT) [inline docs/routing/special_routes.md:45]:
```suggestion
- More about [5XX errors handling](../global_errors_handling.md#5xx-errors-unexpected-errors).
```

File: docs/routing/special_routes.md (hunk 3 above)
Comment (StyleT) [inline docs/routing/special_routes.md:37]: We should explain how to configure 5XX error pages for default/specific domains

File: docs/routing/special_routes.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+Let's go to the `/wrapper/blablabla/` route, ILC will render [404 Error](../global_errors_handling.md#404-error-not-found), because the specified route does not exactly match our routes, аnd all other routes (like `*`) will be ignored (in this case `navbar` it's part of [404 Error](../global_errors_handling.md#404-error-not-found), **not** our route `*` with `Order pos - 100`).
+
+![ILC registry route to error](../assets/routes/route.png)
+
+**Result:**
+
+![ILC registry 404 error example](../assets/routes/fourth-case-result.png)
+
+## '500' Error
```
Comment (StyleT) [inline docs/routing/special_routes.md:35]:
```suggestion
## 5XX Errors
```

File: docs/routing/route_configuration_options.md [hunk 6]
```diff
@@ -1,87 +1,87 @@
 # Route configuration options
```
Comment (StyleT) [inline docs/routing/route_configuration_options.md:1]: Pls extract these changes to another PR. Try to have standalone PRs for every change/fix you do

File: docs/routing/special_routes.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
+# Special routes
+
+To see special routes, use the `Show special` switcher at the top of the routes page
+
+It is used to render errors, if an application with the `Primary` or `Essential` type falls on the page, ILC will not find a match between the address bar and `Route`, `Template` will not be specified in the route chain, etc., ILC will use `Special route` or simple template with an error, depending on the situation:
```
Comment (nightnei) [inline docs/routing/special_routes.md:5]: on the page, FOR EXAMPLE when ILC will not find

Comment (nightnei) [inline docs/routing/special_routes.md:5]: ...`Primary` or `Essential` type falls on the page....
I think more corrent way is ".......`Primary` or `Essential` type is failed on the page......."

File: docs/routing/special_routes.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+## '404' Error
+
+![ILC registry special routes switcher](../assets/routes/special-routes-switcher.png)
+
+- This error usually means that the requested resource was not found on the server.
```
Comment (nightnei) [inline docs/routing/special_routes.md:11]: What do you mean?
404 is rendered only if there is no matched route in ILC Registry.

File: docs/routing/special_routes.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+- When using `Special route` all standard routes are ignored, including `*` with any `Order pos`.
```
Comment (nightnei) [inline docs/routing/special_routes.md:13]: I got what do you want to say here.
But I would say something like `Unlike simple route which uses composition, the special route has own a full configuration. So if you declared, for example, some header by default for any route and want to see it at 404 special route, then you should declare header personally for 404 special route too.`

File: docs/routing/special_routes.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+- If you want something else to be displayed on the page besides the error (header, footer, etc.), you need to add them to the [Slot](./route_configuration_options.md#slot-configuration) as in the screenshot below (This is necessary because standard routes are ignored).
```
Comment (nightnei) [inline docs/routing/special_routes.md:15]: oh, you have additional information here regarding `ignorring`. it's better to combine this two points to one, like I did

File: docs/routing/special_routes.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+## '5XX' Error
+
+This [error](../global_errors_handling.md) occurs if the ILC is unable to process the request (for example, because no [template](./route_configuration_options.md#route_template) is specified in the route chain, or other problems related to the inability of ILC to render the page).
+
+![ILC registry 500 Error](../assets/routes/500-error.png)
+
+- This error is a simple HTML template.
+
+- It is designed in such a way that ILC can render it under any conditions.
```
Comment (nightnei) [inline docs/routing/special_routes.md:43]: combine this point with the previous, since the second point is the explanation for the first one.
and emphasize that it's a simple string(HTML) to avoid problems with rendering it if something went wrong etc

File: docs/routing/special_routes.md [hunk 12]
```diff
@@ -2,17 +2,17 @@
...
+It is used to render errors, if an application with the `Primary` or `Essential` type is failed on the page, for example, when ILC will not find a match between the address bar and `Route`, `Template` will not be specified in the route chain, etc., ILC will use `Special route` or simple template with an error, depending on the situation:
 
 ## '404' Error
 
 ![ILC registry special routes switcher](../assets/routes/special-routes-switcher.png)
 
-- This error usually means that the requested resource was not found on the server.
+- This error can appear in two cases:
+  1. If there is no route configured in Registry for requested URL.
+  2. The app fails to find the requested resource by it's ID (this rule is only for **primary** apps).
```
Comment (nightnei) [inline docs/routing/special_routes.md:13]: 2. what do you mean?
maybe I forgot about this case, hmm

Comment (DammneD) [reply docs/routing/special_routes.md:13]: Got info from here: https://github.com/namecheap/ilc/blob/master/docs/global_errors_handling.md#404-error-not-found

Comment (nightnei) [reply docs/routing/special_routes.md:13]: got it... ok, let it be
but add something like - "more detailed information is in this doc[link]"

**Review by nightnei — APPROVED** [review]

File: docs/routing/special_routes.md (hunk 1 above)
Comment (StyleT) [reply docs/routing/special_routes.md]: Still relevant

File: docs/routing/special_routes.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+## '404' Error
+
+![ILC registry special routes switcher](../assets/routes/special-routes-switcher.png)
+
+- This error can appear in two cases:
+  1. If there is no route configured in Registry for requested URL.
+  2. The app fails to find the requested resource by it's ID (this rule is only for **primary** apps). More details about it [here](../global_errors_handling.md#404-error-not-found)
+
+- When using `Special route` all standard routes are ignored, including `*` with any `Order pos`. If you want something else to be displayed on the page besides the error (header, footer, etc.), you need to add them to the [Slot](./route_configuration_options.md#slot-configuration) as in the screenshot below (This is necessary because standard routes are ignored).
```
Comment (StyleT) [inline docs/routing/special_routes.md:15]: This statement isn't clear. There is no way to "use" special route. 

It would be correct to say: Special routes disregard all of the previously matched routes with "next: true" specified. It means that you always need to specify all necessary slots & template here and shouldn't rely on the global wildcard routes.

File: docs/routing/special_routes.md [hunk 14]
```diff
@@ -12,7 +12,7 @@ It is used to render errors, if an application with the `Primary` or `Essential`
   1. If there is no route configured in Registry for requested URL.
   2. The app fails to find the requested resource by it's ID (this rule is only for **primary** apps). More details about it [here](../global_errors_handling.md#404-error-not-found)
 
-- When using `Special route` all standard routes are ignored, including `*` with any `Order pos`. If you want something else to be displayed on the page besides the error (header, footer, etc.), you need to add them to the [Slot](./route_configuration_options.md#slot-configuration) as in the screenshot below (This is necessary because standard routes are ignored).
+-`Special route` ignores all standart routes, including `*` with any `Order pos`. This means that you must add all the necessary applications (header, footer, etc.) to the `Special route` [slot](./route_configuration_options.md#slot-configuration) (as in the screenshot below).
```
Comment (nightnei) [inline docs/routing/special_routes.md]: As for me this part  "including `*` with any `Order pos`" a little bit confuses.
What do you think about replacing:
-`Special route` ignores all standart routes, including `*` with any `Order pos`.`

With this one:
`Unlike simple routes which use composition with the help of `next: true|false` property , the special route has own full configuration.`

Comment (nightnei) [inline docs/routing/special_routes.md]: add information about necessity of adding `template`

**Review by StyleT — CHANGES_REQUESTED** [review]

File: docs/routing/special_routes.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+
+![ILC registry special routes switcher](../assets/routes/special-routes-switcher.png)
+
+Unlike simple routes which use composition with the help of `Next: true|false` property, the special route has own full configuration. This means that you must add template and all the necessary applications (header, footer, etc.) to the `Special route` [slot](./route_configuration_options.md#slot-configuration) (as in the screenshot below).
+
+![ILC registry error slot](../assets/routes/error-slot.png)
+
+### Example:
+
+Let's go to the `/wrapper/blablabla/` route, ILC will go through the entire list of routes and there will be no required route with a `Next: false` field, so ILC will render [404 Error](../global_errors_handling.md#404-error-not-found).
```
Comment (nightnei) [inline docs/routing/special_routes.md:22]: Too short and that's why unclear. Please explain it in a few sentences.

**Review by nightnei — APPROVED** [review]

**Review by StyleT — APPROVED** [review]

File: docs/routing/special_routes.md [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+
+More about [404 Error here](../global_errors_handling.md#404-error-not-found).
+
+![ILC registry special routes switcher](../assets/routes/special-routes-switcher.png)
+
+Unlike simple routes which use composition with the help of `Next: true|false` property, the special route has own full configuration. This means that you must add template and all the necessary applications (header, footer, etc.) to the `Special route` [slot](./route_configuration_options.md#slot-configuration) (as in the screenshot below).
+
+![ILC registry error slot](../assets/routes/error-slot.png)
+
+### Example:
```
Comment (StyleT) [inline docs/routing/special_routes.md:20]: Replace this section with a link to https://github.com/namecheap/ilc/blob/e6e5809f2d9e90de1db3c3b4ff5bce8a37fdf4d1/docs/global_errors_handling.md#404-error-not-found

File: docs/routing/special_routes.md [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+
+Let's go to the `/wrapper/blablabla/` route, ILC requires the route chain to end with a route with the `Next` field parameter: `false`, in our case we don’t have a suitable route at all, which means that ILC will go through the entire list of routes and not finding the closing route (with the `Next: false` parameter) and so it will render a [404 Error](../global_errors_handling.md#404-error-not-found).
+
+![ILC registry route to error](../assets/routes/route.png)
+
+**Result:**
+
+![ILC registry 404 error example](../assets/routes/fourth-case-result.png)
+
+## '5XX' Error
```
Comment (StyleT) [inline docs/routing/special_routes.md:30]:
```suggestion
## '5XX' Errors
```

File: docs/routing/special_routes.md [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+
+This error is a simple HTML template. It is designed in such a way to not use ILC routing unlike 404 errors, which means that even if something went wrong with ILC, this error will be rendered.
```
Comment (StyleT) [inline docs/routing/special_routes.md:32]: 1. Error couldn't be called a "template". Pls rephrase.
2. It wasn't designed to not to use ILC routing. It's was designed to be a static markup in order to minimise chances of error occurrence during it's rendering. As it's a last chance to show something meaningful to visitor.

File: docs/routing/special_routes.md [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+![ILC registry domains menu](../assets/routes/domain-create.png)
+![ILC registry set template of 500 error](../assets/routes/set-template-for-domains.png)
+
+### Example:
+
+In ILC it is possible to check how your 500 error is displayed by going to the route `/_ilc/500/`. Example of the route at our demo website - http://demo.microfrontends.online/_ilc/500
+
+Now let's move on to a real example of a case in which we will render a 500 error:
+
+Let's go to the `/notemplate/` route, in this case, no template for the route is specified, so ILC cannot render the application, in which case we will see a [500 error](../global_errors_handling.md#5xx-errors-unexpected-errors).
```
Comment (StyleT) [inline docs/routing/special_routes.md:50]: 1. There are no such route at demo website
2. Replace all starting from `Now let's move on to a real example` with a link https://github.com/namecheap/ilc/blob/e6e5809f2d9e90de1db3c3b4ff5bce8a37fdf4d1/docs/global_errors_handling.md#5xx-errors-unexpected-errors

**Review by StyleT — CHANGES_REQUESTED** [review]

**Review by bandantonio — CHANGES_REQUESTED** [review]
@DammneD Please make sure that the document in this PR conforms to the unified approach introduced in the #363

File: docs/routing/special_routes.md [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+Unlike simple routes which use composition with the help of `Next: true|false` property, the special route has own full configuration. This means that you must add template and all the necessary applications (header, footer, etc.) to the `Special route` [slot](./route_configuration_options.md#slot-configuration) (as in the screenshot below).
+
+![ILC registry error slot](../assets/routes/error-slot.png)
+
+More about [404 Error here](../global_errors_handling.md#404-error-not-found).
+
+## '5XX' Errors
+
+This error is created as a simple HTML template. It is designed in such a way to be a static markup in order to minimise chances of error occurrence during it's rendering. As it's a last chance to show something meaningful to visitor.
+The template with the name: `500` is used as the default error template for the main domain.
```
Comment (bandantonio) [inline docs/routing/special_routes.md:23]:
```suggestion
The '500' template acts as the default error template for the main domain.
```

File: docs/routing/special_routes.md [hunk 21]
```diff
@@ -0,0 +1,36 @@
...
+
+![ILC registry 500 Error](../assets/routes/500-error.png)
```
Comment (bandantonio) [inline docs/routing/special_routes.md:25]:
```suggestion
![ILC registry 500 error](../assets/routes/500-error.png)
```

File: docs/routing/special_routes.md [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+
+To assign different error templates for different domains, you must specify the template when creating the domain, as shown in the screenshots below.
```
Comment (bandantonio) [inline docs/routing/special_routes.md:27]:
```suggestion
To assign different error templates to domains, you must specify the template when creating a domain. See the screenshots below.
```

File: docs/routing/special_routes.md [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+
+![ILC registry domains menu](../assets/routes/domain-create.png)
+![ILC registry set template of 500 error](../assets/routes/set-template-for-domains.png)
+
+
+    ??? tip "Check what your error looks like"
+        In ILC it is possible to check how your 500 error is displayed by going to the route `/_ilc/500/`. Example of the route at our demo website - http://demo.microfrontends.online/_ilc/500
```
Comment (bandantonio) [inline docs/routing/special_routes.md:34]:
```suggestion
??? tip "Preview error"
    In ILC, it is possible to preview the 500 error via the `/_ilc/500/` route.

    For example: `http://demo.microfrontends.online/_ilc/500` is the preview route for the demo website.
```

File: docs/routing/special_routes.md [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+
+More about [5XX errors handling](../global_errors_handling.md#5xx-errors-unexpected-errors).
```
Comment (bandantonio) [inline docs/routing/special_routes.md:36]:
```suggestion
More information about the 5xx errors handling is available in the [Global error handling page](../global_errors_handling.md#5xx-errors-unexpected-errors).
```

File: docs/routing/special_routes.md [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+This error is created as a simple HTML template. It is designed in such a way to be a static markup in order to minimise chances of error occurrence during it's rendering. As it's a last chance to show something meaningful to visitor.
+The template with the name: `500` is used as the default error template for the main domain.
+
+![ILC registry 500 Error](../assets/routes/500-error.png)
+
+To assign different error templates for different domains, you must specify the template when creating the domain, as shown in the screenshots below.
+
+![ILC registry domains menu](../assets/routes/domain-create.png)
+![ILC registry set template of 500 error](../assets/routes/set-template-for-domains.png)
+
```
Comment (bandantonio) [inline docs/routing/special_routes.md:31]:
```suggestion
```

**Review by bandantonio — CHANGES_REQUESTED** [review]
